src/components/structures/auth/Registration.js

```diff
@@ -64,6 +64,15 @@ module.exports = React.createClass({
 
     getInitialState: function() {
         const customURLsAllowed = !SdkConfig.get()['disable_custom_urls'];
+        let initialPhase = PHASE_SERVER_DETAILS;
+        if (
+            // if we have these two, skip to the good bit
+            (this.props.clientSecret && this.props.sessionId) ||
```

Comment:
Please add a note that these can come in from email verification, as a reminder for the future.